galley/pkg/config/schema/schema.go

```diff
@@ -127,7 +127,10 @@ type KubeResources []KubeResource
 
 // CanonicalResourceName of the resource.
 func (i KubeResource) CanonicalResourceName() string {
-	return fmt.Sprintf("%s.%s/%s", i.Group, i.Version, i.Kind)
+	if i.Group == "" {
+		return fmt.Sprintf("core/%s/%s", i.Version, i.Kind)
```

Comment:
nit: Sprintf is really slow ... I'm assuming a method like this might get called often. Consider basic string concatenation.

Reply:
This is used for rare logging. Still switching to concat is easy. :)